pallets/corporate-actions/src/distribution/mod.rs

```diff
@@ -185,6 +188,7 @@ decl_module! {
             ca_id: CAId,
             portfolio: Option<PortfolioNumber>,
             currency: Ticker,
+            per_share: T::Balance,
```

Comment:
Does it make  sense a `per_share` == 0? If not we should `ensure` that or enforce it by its type.

Reply:
Interesting. I wonder if there is a jurisdiction where companies have to record dividend payments even if the amount being paid is zero.

In any case, since I don't see any real downside of this. The issuer can just start a new dividend if they mess us this var. Therefore, I think it's best that we don't add this check.